roles/nova-data/tasks/v7k.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: install multipath-tools packages (ubuntu)
+  package: name={{ item }}
+  with_items:
+    -  multipath-tools
+  when: ursula_os == 'ubuntu'
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: install device-mapper-multipath packages
+  package: name={{ item }}
+  with_items:
+    - device-mapper-multipath
+  when: ursula_os == 'rhel'
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: generate multipath.conf
+  template: src=etc/multipath.conf dest=/etc/multipath.conf mode=0640
+            owner=root group=root
+
+- name: restart multipath-tools
+  service:
+    name:  multipath-tools
+    state: restarted
+  when: ursula_os == 'ubuntu'
```

Comment:
U can group above tasks in 2 blocks (1 for ubuntu and 2nd for rhel)
- block:
   -task1
   -taks2
   -taks3
  when: ursula_os == 'ubuntu'